### PR TITLE
Laravel 9 compatibility

### PR DESCRIPTION
**Pull request by alexchirolde**

### CODE REVIEW

**Comment by ctf0** [comment]
thank you, i will take a look at it today.

File: src/App/Controllers/Modules/GetContent.php [hunk 1]
```diff
@@ -94,11 +94,24 @@ protected function getData($dir)
...
+            return [
+                'path' => $path,
+                'dirname' => str_replace("/$name", '', $path),
+                'basename' => $name,
+                'extension' => $extension,
+                'filename' => substr($name, 0, strrpos($name, '.')),
+                'timestamp' => $attributes->lastModified(),
+                'size' => $attributes->isFile() ? $attributes->fileSize() : 0,
+                'type' => $attributes->isFile() ? 'file' : 'dir',
+                'mimetype' => $attributes->isFile() ? ($attributes->mimeType() ?: $this->getMimeFromExtension($extension)) : 'folder',
```
Comment (ctf0) [inline src/App/Controllers/Modules/GetContent.php]: for this we can use https://packagist.org/packages/league/mime-type-detection which is a dep of flysystem https://github.com/thephpleague/flysystem/blob/e5d33a0dd64f486da72e27b19091e46248f71a66/composer.json#L21

File: composer.json [hunk 2]
```diff
@@ -9,17 +9,25 @@
...
+        },
+        {
+            "name": "Alejandro T",
+            "email": "alexchirolde@gmail.com"
         }
     ],
     "require": {
-        "illuminate/support": ">=7.0 <9.0",
-        "maennchen/zipstream-php": ">=1.0",
+        "illuminate/support": "^9.14.1",
```
Comment (ctf0) [inline composer.json]: plz change to `^9.0` to match all v9 versions

File: src/App/Controllers/Modules/GetContent.php [hunk 3]
```diff
@@ -94,11 +94,24 @@ protected function getData($dir)
      */
     protected function getFolderContent($folder, $rec = false)
     {
-        return $this->storageDisk->listWith(
-            ['mimetype', 'visibility', 'timestamp', 'size'],
-            $folder,
-            $rec
-        );
+        return $this->storageDisk->listContents($folder)->map(function ($attributes) {
```
Comment (ctf0) [inline src/App/Controllers/Modules/GetContent.php]: plz add `$rec` to the `listContents` check https://flysystem.thephpleague.com/docs/usage/filesystem-api/

**Comment by alexchirolde** [comment]
Check it now @ctf0

File: src/App/Controllers/Modules/GetContent.php [hunk 4]
```diff
@@ -94,11 +94,24 @@ protected function getData($dir)
...
+                'path' => $path,
+                'dirname' => str_replace("/$name", '', $path),
+                'basename' => $name,
+                'extension' => $extension,
+                'filename' => substr($name, 0, strrpos($name, '.')),
+                'timestamp' => $attributes->lastModified(),
+                'size' => $attributes->isFile() ? $attributes->fileSize() : 0,
+                'type' => $attributes->isFile() ? 'file' : 'dir',
+                'mimetype' => $attributes->isFile() ? ($this->storageDisk->mimeType($path)) : 'folder',
+                'etag' => $attributes->extraMetadata()['ETag'] ?? null,
```
Comment (ctf0) [inline src/App/Controllers/Modules/GetContent.php]: do we need this ?

Comment (alexchirolde) [reply src/App/Controllers/Modules/GetContent.php]: With this we format the information per folder/file in the FileManager, pretty much something like:

`return $this->storageDisk->listWith(
            ['mimetype', 'visibility', 'timestamp', 'size'],
            $folder,
            $rec
        );`

Comment (ctf0) [reply src/App/Controllers/Modules/GetContent.php]: i meant the etag, i cant recall that i used it b4, maybe i forgot but if there's no use for it then we can remove it.

**Comment by alexchirolde** [comment]
@ctf0 when will this PR be available on the main branch? I kind of need it ;-)

**Comment by ctf0** [comment]
i will accept it but will need to do some tests b4 publishing it, but for now u can install it using dev-main, thanks again for ur help 🙏

**Comment by alexchirolde** [comment]
Thxs to you @ctf0, glad to help.

**Pull request merged**